cpp/tests/test_ide_secir_ageres.cpp

```diff
@@ -0,0 +1,159 @@
+    vec_delaydistrib2[(int)mio::isecir::InfectionTransition::SusceptibleToExposed].set_distribution_parameter(-1.);
+    vec_delaydistrib2[(int)mio::isecir::InfectionTransition::InfectedNoSymptomsToInfectedSymptoms]
+        .set_distribution_parameter(2.0);
+
+    model.parameters.get<mio::isecir::TransitionDistributions>()[mio::AgeGroup(1)] = vec_delaydistrib2;
+
+    // Third AgeGroup for TransitionDistributions.
+    mio::SmootherCosine smoothcos3(2.5);
+    mio::StateAgeFunctionWrapper delaydistribution3(smoothcos3);
+    std::vector<mio::StateAgeFunctionWrapper> vec_delaydistrib3(num_transitions, delaydistribution3);
+    // TransitionDistribution is not used for SusceptibleToExposed. Therefore, the parameter can be set to any value.
+    vec_delaydistrib1[(int)mio::isecir::InfectionTransition::SusceptibleToExposed].set_distribution_parameter(-1.);
+    vec_delaydistrib1[(int)mio::isecir::InfectionTransition::InfectedNoSymptomsToInfectedSymptoms]
+        .set_distribution_parameter(4.0);
+
+    model.parameters.get<mio::isecir::TransitionDistributions>()[mio::AgeGroup(2)] = vec_delaydistrib3;
+
+    //TransitionProbabilities.
+    for (auto group = mio::AgeGroup(0); group < mio::AgeGroup(num_agegroups); ++group) {
+        std::vector<ScalarType> vec_prob(num_transitions, 0.5);
+        // The following probabilities must be 1, as there is no other way to go.
+        vec_prob[Eigen::Index(mio::isecir::InfectionTransition::SusceptibleToExposed)]        = 1;
+        vec_prob[Eigen::Index(mio::isecir::InfectionTransition::ExposedToInfectedNoSymptoms)] = 1;
+        model.parameters.get<mio::isecir::TransitionProbabilities>()[group]                   = vec_prob;
+    }
+
+    // Contact matrix.
+    mio::ContactMatrixGroup contact_matrix = mio::ContactMatrixGroup(1, num_agegroups);
+    contact_matrix[0] = mio::ContactMatrix(Eigen::MatrixXd::Constant(num_agegroups, num_agegroups, 10.));
+    model.parameters.get<mio::isecir::ContactPatterns>() = mio::UncertainContactMatrix(contact_matrix);
+
+    mio::ExponentialSurvivalFunction exponential(0.5);
+    mio::StateAgeFunctionWrapper prob(exponential);
+    for (auto group = mio::AgeGroup(0); group < mio::AgeGroup(num_agegroups); ++group) {
+        model.parameters.get<mio::isecir::TransmissionProbabilityOnContact>()[group] = prob;
+        model.parameters.get<mio::isecir::RelativeTransmissionNoSymptoms>()[group]   = prob;
+        model.parameters.get<mio::isecir::RiskOfInfectionFromSymptomatic>()[group]   = prob;
+    }
+    model.parameters.set<mio::isecir::Seasonality>(0.1);
+    // Start the simulation on the 40th day of a year (i.e. in February).
+    model.parameters.set<mio::isecir::StartDay>(40);
+
+    model.check_constraints(dt);
+
+    //Compare compartments with previous run.
+    mio::TimeSeries<ScalarType> compartments = simulate(tmax, dt, model);
+    auto compare_compartments                = load_test_data_csv<ScalarType>("ide-secir-ageres-compare.csv");
```

Comment:
```suggestion
    auto compare_compartments                << ..
```
Can you copy here the entries for the last time point from "ide-secir-ageres-compare.csv", similar as in https://github.com/SciCompMod/memilio/blob/main/cpp/tests/test_lct_secir.cpp#L323? Then we can compare the results at the last time point.